### PR TITLE
Cleaning up matter initialization.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
## Description of changes
- Moves `material` and `set_material()` down to `/obj`.
- Cleans up some Initialize() overrides on `/obj/item` and `/obj/structure`.
- Reworks some matter gen to make recipes less brittle.

## Why and what will this PR improve
Fixes issues with the inconsistent material from crafting recipes unit test, works towards standardized material handling.

## Authorship
Myself and @PsyCommando.

## Changelog
Nothing player-facing.

### CODE REVIEW

File: code/game/machinery/_machines_base/stock_parts/network_receiver.dm [hunk 1]
```diff
@@ -6,7 +6,7 @@
 	icon_state = "net_lock"
 	part_flags = PART_FLAG_QDEL
 
-/obj/item/stock_parts/network_receiver/Initialize(ml, material_key)
+/obj/item/stock_parts/network_receiver/Initialize()
```
Comment (PsyCommando) [inline code/game/machinery/_machines_base/stock_parts/network_receiver.dm:9]: If you remove those the compiler won't let you refer to them as named parameters with subclasses. And the auto-complete won't list them as parameters. It really is not a good idea to remove them imo.

Comment (MistakeNot4892) [reply code/game/machinery/_machines_base/stock_parts/network_receiver.dm:9]: I'm removing them so I can mass find and replace Initialize() later.

Comment (PsyCommando) [reply code/game/machinery/_machines_base/stock_parts/network_receiver.dm:9]: Alright then.

**Comment by MistakeNot4892** [comment]
Will revisit these down the track.

**Pull request closed without merging**